core/db/db.go

```diff
@@ -73,10 +74,22 @@ func Create() *OpenchainDB {
 
 // GetDBHandle get an opened openchainDB singleton
 func GetDBHandle() *OpenchainDB {
-	openchainDB.Open()
 	return openchainDB
 }
 
+// Start the db
+func Start() {
+	once.Do(func() {
+		openchainDB = Create()
+	})
+	openchainDB.Open()
+}
+
+// Stop the db
+func Stop() {
+	openchainDB.Close()
+}
+
```

Comment:
I think that both pairs of methods (i.e., start/stop and open/close) are confusing. Can you please
 - make the methods open/close private and use start/stop outside this file (including the files db_test.go and db_test_exports.go).
 - Document that now method `GetDBHandle()` requires that db.Start() should have been called before. In other words, method `GetDBHandle()` and `Start()` are not mutually thread-safe.
- Now (see above point) locks inside `Open()` and `Close()` does not make sense and can be removed. Because, `GetHandle()` is without locks (I guess, this is what you intent to achieve). In fact, this is how it was before you had introduced locks in `Open()` and `Close()` :-)